### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Install dependencies
+      run: >
+        sudo apt-get update &&
+        sudo ./contrib/ci/fwupd_setup_helpers.py install-dependencies --yes -o ubuntu &&
+        python3 -m pip install --user "meson >= 0.60.0"
```
[the remark this answers is not in the document]
Comment (hughsie) [reply .github/workflows/codeql-analysis.yml:38]: Ya, Ubuntu having an old meson tripped up quite a few people.

[the remark this answers is not in the document]
Comment (hughsie) [reply .github/workflows/codeql-analysis.yml:38]: Lets fix this async to this PR; the LGTM results failing (or at least not passing) is preventing anyone now a maintainer from merging PRs...

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '28 5 * * 6'
+
```
Comment (superm1) [inline .github/workflows/codeql-analysis.yml:10]: does this take too long that it can't run on every PR?

Comment (hughsie) [reply .github/workflows/codeql-analysis.yml:10]: It's 9 minutes; 2 minutes longer than fedora build and one less than the fuzzer. I'll delete it and we can add it back if we need to.

**Pull request merged**